File: Ahorn/entities/dreamSpinnerFake.jl
module IsaGrabBagDreamSpinnerFake

using ..Ahorn, Maple

@mapdef Entity "isaBag/dreamSpinFake" DreamSpinnerFake(x::Integer, y::Integer)

const placements = Ahorn.PlacementDict(
	"Dream Spinner Fake (IsaGrabBag)" => Ahorn.EntityPlacement(
		DreamSpinnerFake
	)
)

function Ahorn.selection(entity::DreamSpinnerFake)
	x, y = Ahorn.position(entity)
	return Ahorn.getSpriteRectangle("isafriend/danger/crystal/fg_dreamspinner_fake.png", x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::DreamSpinnerFake, room::Maple.Room)
	Ahorn.drawSprite(ctx, "isafriend/danger/crystal/fg_dreamspinner_fake.png", 0, 0)
end

end

File: Loenn/entities/dream_spinner_fake.lua
local utils = require("utils")

local dreamSpinnerFake = {}

dreamSpinnerFake.name = "isaBag/dreamSpinFake"
dreamSpinnerFake.depth = -8500
dreamSpinnerFake.placements = {
    name = "default",
}
dreamSpinnerFake.texture = "isafriend/danger/crystal/fg_dreamspinner_fake"

function dreamSpinnerFake.selection(room, entity)
    return utils.rectangle(entity.x - 10, entity.y - 10, 21, 21)
end

return dreamSpinnerFake